openmdao/core/component.py

```diff
@@ -511,13 +573,34 @@ def apply_nonlinear(self, params, unknowns, resids):
         # cache the old values of the unknowns.
         resids.vec[:] = -unknowns.vec
 
-        self.solve_nonlinear(params, unknowns, resids)
+        self._sys_solve_nonlinear(params, unknowns, resids)
 
         # Unknowns are restored to the old values too. apply_nonlinear does
         # not change the output vector.
         resids.vec[:] += unknowns.vec
         unknowns.vec[:] -= resids.vec
 
+    def _sys_solve_nonlinear(self, params, unknowns, resids):
+        """
+        Runs the component. The user is required to define this function in
```

Comment:
I think "this" refers to solve_nonlinear?  It's unclear... suggest replacing "this" with the function name...   similar comment for _sys_apply_nonlinear docstring